tools/autograd/gen_python_functions.py

```diff
@@ -186,7 +189,10 @@ def should_bind(declaration):
 
     py_torch_functions = group_declarations_by_name(declarations, should_bind)
 
-    env = create_python_bindings(py_torch_functions, has_self=False)
+    if PY2:
```

Comment:
[`PyModule_AddFunctions`](https://docs.python.org/3/c-api/module.html#c.PyModule_AddFunctions) don't exists in PY2.
generated codes conditionally with `#if PY_MAJOR_VERSION == 2`.